### PR TITLE
expand to return measurement dict

### PR DESCRIPTION
**Pull request by jtwang1027**
Expand ProLIF `detector` methods to return a "measurement dict" that contains additional info : distance, dihedral angles, plane angle, etc.

This result gets returned alongside ligand and protein atom index when `fp.to_dataframe( return_atoms=True)` (which is usually called after `fp.run` or `fp.run_from_iterable`.

This has been tested with a few MD trajectories as well as docked ligand/protein systems.

### CODE REVIEW

File: prolif/utils.py [hunk 1]
```diff
@@ -247,14 +253,15 @@ def to_dataframe(ifp, interactions, index_col="Frame", dtype=None,
...
+        # presents tuple of ligand atom idx, protein atom idx, and measurements_dict
         df = (df.groupby(axis=1, level=["ligand", "protein", "interaction"])
                 .agg(tuple))
     if dtype:
         df = df.astype(dtype)
     if drop_empty:
         if has_atom_indices and return_atoms:
             mask = df.apply(lambda s:
-                            ~(s.isin([(None, None)]).all()), axis=0)
+                            ~(s.isin([(None, None, None)]).all()), axis=0)
```
Comment (jtwang1027) [inline prolif/utils.py:264]: 'drop_empty' drops rows that don't have any interactions. Normally, it looks for `(None, None)` bc it's (lig_idx, prot_idx). 

Now, it needs to look for `(None, None,None)` (lig_idx, prot_idx, measurement_dict)

File: prolif/interactions.py [hunk 2]
```diff
@@ -168,21 +170,23 @@ def detect(self, acceptor, donor):
...
+                            {'distance':round(dist, MEASUREMENT_DECIMAL_PLACES),
+                             'angle':round(angle*180/PI, MEASUREMENT_DECIMAL_PLACES)}
+        return False, None, None, None
 
 
 class HBDonor(_BaseHBond):
     """Hbond interaction between a ligand (donor) and a residue (acceptor)"""
     def detect(self, ligand, residue):
-        bit, ires, ilig = super().detect(residue, ligand)
-        return bit, ilig, ires
```
Comment (dangthatsright) [inline prolif/interactions.py:185]: the return index order is swapped here, does this matter at all?

File: prolif/interactions.py [hunk 3]
```diff
@@ -258,8 +266,7 @@ def detect(self, ligand, residue):
 class XBDonor(_BaseXBond):
     """Halogen bonding between a ligand (donor) and a residue (acceptor)"""
     def detect(self, ligand, residue):
-        bit, ires, ilig = super().detect(residue, ligand)
-        return bit, ilig, ires
```
Comment (dangthatsright) [inline prolif/interactions.py:262]: same here

File: prolif/interactions.py [hunk 4]
```diff
@@ -280,8 +287,7 @@ def detect(self, ligand, residue):
 class Anionic(_BaseIonic):
     """Ionic interaction between a ligand (anion) and a residue (cation)"""
     def detect(self, ligand, residue):
-        bit, ires, ilig = super().detect(residue, ligand)
-        return bit, ilig, ires
```
Comment (dangthatsright) [inline prolif/interactions.py:284]: and here

File: prolif/interactions.py [hunk 5]
```diff
@@ -332,16 +339,16 @@ def detect(self, cation, pi):
...
+                            {'distance':round(dist,MEASUREMENT_DECIMAL_PLACES),
+                            'angle':round(angle*180/PI, MEASUREMENT_DECIMAL_PLACES)}
+        return False, None, None, None
 
 class PiCation(_BaseCationPi):
     """Cation-Pi interaction between a ligand (aromatic ring) and a residue
     (cation)"""
     def detect(self, ligand, residue):
-        bit, ires, ilig = super().detect(residue, ligand)
-        return bit, ilig, ires
```
Comment (dangthatsright) [inline prolif/interactions.py:344]: and here

File: prolif/interactions.py [hunk 6]
```diff
@@ -452,8 +461,7 @@ def detect(self, ligand, residue):
 class MetalAcceptor(_BaseMetallic):
     """Metallic interaction between a ligand (chelated) and a metal residue"""
     def detect(self, ligand, residue):
-        bit, ires, ilig = super().detect(residue, ligand)
-        return bit, ilig, ires
```
Comment (dangthatsright) [inline prolif/interactions.py:456]: another one

**Review by dangthatsright — CHANGES_REQUESTED** [review]
had one question on the return values of various interaction classes. it seems like it previously assumed you returned ligand index, protein index - i'm assuming this is important?

**Comment by jtwang1027** [comment]
> had one question on the return values of various interaction classes. it seems like it previously assumed you returned ligand index, protein index - i'm assuming this is important?

good catch, i'll switch them back

**Review by dangthatsright — APPROVED** [review]
cool lgtm

**Pull request merged**